src/dotty/tools/dotc/core/tasty/TreePickler.scala

```diff
@@ -217,8 +217,8 @@ class TreePickler(pickler: TastyPickler) {
     case tpe: RefinedType =>
       writeByte(REFINEDtype)
       withLength {
-        pickleType(tpe.parent)
```

Comment:
To harmonize the layout. Non-trees now always come before trees in Tasy entries (except for some cases where trailing non-trees are optional as explained in numRefs.